nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java

```diff
@@ -566,9 +570,10 @@ private void handleEnhancedForOverKeySet(
             AccessPath.mapWithIteratorContentsKey(mapNode, lhs, apContext);
         if (mapWithIteratorContentsKey != null) {
           // put sanity check here to minimize perf impact
-          if (!isCallToMethod(rhsInv, "java.util.Set", "iterator")) {
+          if (!isCallToMethod(rhsInv, SET_TYPE_SUPPLIER, "iterator")) {
```

Comment:
This is a key change; we allow `iterator()` to come from a subtype of `Set` (since, e.g., `ImmutableMap.keySet()` returns an `ImmutableSet`)